homeassistant/components/yale_smart_alarm/alarm_control_panel.py

```diff
@@ -20,92 +19,124 @@
 )
 from homeassistant.const import (
     CONF_NAME,
-    CONF_PASSWORD,
     CONF_USERNAME,
     STATE_ALARM_ARMED_AWAY,
     STATE_ALARM_ARMED_HOME,
     STATE_ALARM_DISARMED,
+    STATE_UNAVAILABLE,
 )
-import homeassistant.helpers.config_validation as cv
+from homeassistant.core import callback
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
 
-CONF_AREA_ID = "area_id"
+from .const import DOMAIN, LOGGER
+from .coordinator import YaleDataUpdateCoordinator
 
-DEFAULT_NAME = "Yale Smart Alarm"
 
-DEFAULT_AREA_ID = "1"
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the alarm entry."""
 
-_LOGGER = logging.getLogger(__name__)
+    async_add_entities(
+        [YaleAlarmDevice(coordinator=hass.data[DOMAIN][entry.entry_id]["coordinator"])]
+    )
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_USERNAME): cv.string,
-        vol.Required(CONF_PASSWORD): cv.string,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Optional(CONF_AREA_ID, default=DEFAULT_AREA_ID): cv.string,
-    }
-)
-
-
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the alarm platform."""
-    name = config[CONF_NAME]
-    username = config[CONF_USERNAME]
-    password = config[CONF_PASSWORD]
-    area_id = config[CONF_AREA_ID]
-
-    try:
-        client = YaleSmartAlarmClient(username, password, area_id)
-    except AuthenticationError:
-        _LOGGER.error("Authentication failed. Check credentials")
-        return
-
-    add_entities([YaleAlarmDevice(name, client)], True)
 
-
-class YaleAlarmDevice(AlarmControlPanelEntity):
+class YaleAlarmDevice(CoordinatorEntity, AlarmControlPanelEntity):
     """Represent a Yale Smart Alarm."""
 
-    def __init__(self, name, client):
-        """Initialize the Yale Alarm Device."""
-        self._name = name
-        self._client = client
-        self._state = None
+    coordinator: YaleDataUpdateCoordinator
 
-        self._state_map = {
-            YALE_STATE_DISARM: STATE_ALARM_DISARMED,
-            YALE_STATE_ARM_PARTIAL: STATE_ALARM_ARMED_HOME,
-            YALE_STATE_ARM_FULL: STATE_ALARM_ARMED_AWAY,
-        }
+    _state_map = {
+        YALE_STATE_DISARM: STATE_ALARM_DISARMED,
+        YALE_STATE_ARM_PARTIAL: STATE_ALARM_ARMED_HOME,
+        YALE_STATE_ARM_FULL: STATE_ALARM_ARMED_AWAY,
+    }
+
+    _state = STATE_UNAVAILABLE
 
     @property
     def name(self):
         """Return the name of the device."""
-        return self._name
+        return self.coordinator.entry.data[CONF_NAME]
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this entity."""
+        return str(self.coordinator.entry.unique_id)
```

Comment:
As I understand https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-of-last-resort you should use `self.coordinater.entry.entry_id` here.

Reply:
You're right. I went to fast. Thanks for checking
Solved in next commit